infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/AsyncStorageProvisioner.java

```diff
@@ -0,0 +1,383 @@
+ * via SSH - configmap, with public part of SSH key - PVC for storing backups;
+ */
+public class AsyncStorageProvisioner {
+
+  private static final int SERVICE_PORT = 2222;
+  /**
+   * The authorized_keys file in SSH specifies the SSH keys that can be used for logging into the
+   * user account for which the file is configured.
+   */
+  private static final String AUTHORIZED_KEYS = "authorized_keys";
+  /**
+   * Name of the asynchronous storage Pod and Service. Rsync command will use this Service name for
+   * communications: e.g.: rsync ${RSYNC_OPTIONS} --rsh="ssh ${SSH_OPTIONS}" async-storage:/{PATH}
+   */
+  static final String ASYNC_STORAGE = "async-storage";
+  /** The name suffix for ConfigMap with SSH configuration */
+  static final String ASYNC_STORAGE_CONFIG = "async-storage-config";
+  /** The path of mount storage volume for file persist */
+  private static final String ASYNC_STORAGE_DATA_PATH = "/" + ASYNC_STORAGE;
+  /** The path to the authorized_keys */
+  private static final String SSH_KEY_PATH = "/.ssh/" + AUTHORIZED_KEYS;
+  /** The name of SSH key pair for rsync */
+  static final String SSH_KEY_NAME = "rsync-via-ssh";
+  /** The name of volume for mounting configuration map and authorized_keys */
+  private static final String CONFIG_MAP_VOLUME_NAME = "async-storage-configvolume";
+  /** */
+  private static final String STORAGE_VOLUME = "async-storage-data";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncStorageProvisioner.class);
+
+  private final String pvcQuantity;
+  private final String storageImage;
+  private final String accessMode;
+  private final String strategy;
+  private final String configuredPVCName;
+  private final String pvcStorageClassName;
+  private final SshManager sshManager;
+  private final OpenShiftClientFactory clientFactory;
+
+  @Inject
+  public AsyncStorageProvisioner(
+      @Named("che.infra.kubernetes.pvc.quantity") String pvcQuantity,
+      @Named("che.infra.kubernetes.async.storage.image") String image,
+      @Named("che.infra.kubernetes.pvc.access_mode") String accessMode,
+      @Named("che.infra.kubernetes.pvc.strategy") String strategy,
+      @Named("che.infra.kubernetes.pvc.name") String configuredPVCName,
+      @Named("che.infra.kubernetes.pvc.storage_class_name") String pvcStorageClassName,
+      SshManager sshManager,
+      OpenShiftClientFactory openShiftClientFactory) {
+    this.pvcQuantity = pvcQuantity;
+    this.storageImage = image;
+    this.accessMode = accessMode;
+    this.strategy = strategy;
+    this.configuredPVCName = configuredPVCName;
+    this.pvcStorageClassName = pvcStorageClassName;
+    this.sshManager = sshManager;
+    this.clientFactory = openShiftClientFactory;
+  }
+
+  public void provision(OpenShiftEnvironment osEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+    if (!parseBoolean(osEnv.getAttributes().get(ASYNC_PERSIST_ATTRIBUTE))) {
+      return;
+    }
+
```

Comment:
do we need `(parseBoolean(osEnv.getAttributes().get(ASYNC_PERSIST_ATTRIBUTE)` condition here?

Reply:
fixed